sdcm/utils/gce_utils.py

```diff
@@ -237,6 +239,71 @@ def gcloud(self) -> GcloudContextManager:
         return GcloudContextManager(self, 'gcloud')
 
 
+class GkeClusterForCleaner:
+    def __init__(self, cluster_info: dict, cleaner: "GkeCleaner"):
+        self.cluster_info = cluster_info
+        self.cleaner = cleaner
+
+    @cached_property
+    def metadata(self) -> dict:
+        metadata = self.cluster_info["nodeConfig"]["metadata"].items()
+        return {"items": [{"key": key, "value": value} for key, value in metadata], }
+
+    @cached_property
+    def name(self) -> str:
+        return self.cluster_info["name"]
+
+    @cached_property
+    def zone(self) -> str:
+        return self.cluster_info["zone"]
+
+    def destroy(self):
+        return self.cleaner.gcloud.run(f"container clusters delete {self.name} --zone {self.zone} --quiet")
+
+
+class GkeCleaner(GcloudContainerMixin):
```

Comment:
I understand this was only moved here, but I'll leave comment for the future: Api of this class is very misleading and complicated.
`GkeCleaner` that doesn't have 'cleaning' methods - mostly listing resources (and e.g. `list_gke_clusters` returns list of `GkeClusterForCleaner` - so again mislead as it does not return gke cluster itself).

When introducing classes with simple goals like cleaning, let's make sure names, methods are consistent with this goal.